src/lib/db.ts

```diff
@@ -0,0 +1,83 @@
+import * as firebase from "firebase";
+import { AppMode } from "../models/stores";
+
+export type IDBConnectOptions = IDBAuthConnectOptions | IDBNonAuthConnectOptions;
+export interface IDBAuthConnectOptions {
+  appMode: "authed";
+  rawFirebaseToken: string;
+}
+export interface IDBNonAuthConnectOptions {
+  appMode: "dev" | "test";
+}
+
+export class DB {
+  private appMode: AppMode;
+  private firebaseUser: firebase.User | null = null;
+
+  public get isConnected() {
+    return this.firebaseUser !== null;
```

Comment:
Assuming undefined should be treated like null, I would use `!=` here.

Reply:
It can't be undefined though, only a string value or null.  I think its better to use explicit checks.